contribute/architecture/frontend-data-requests.md

```diff
@@ -0,0 +1,51 @@
+# Data requests
+
+[BackendSrv is the core data request service](https://grafana.com/docs/grafana/latest/packages_api/runtime/backendsrv) used by Grafana. This documentation highlights the high level concepts used by `BackendSrv`.
+
+## Request `cancellation`
+> **Note:** Data sources can implement their own cancellation concept. This documentation describes the `request cancellation` included in Grafana.
+
+### The challenge
+A data request can take a long time to finish. Between the request starts and finishes, the user might change context. For instance, the user navigates away or the user issues the same request again.
+
+If we allow ongoing requests to continue, it might cause unnecessary load on data sources.
+
+Grafana uses a concept referenced as `request cancellation` to cancel any ongoing request that Grafana doesn’t need.
+
+#### Before `Grafana 7.2`
+Before Grafana can cancel any data request, Grafana needs to have a way to identify each request. Grafana identifies each request with the property `requestId` [passed as options](https://github.com/grafana/grafana/blob/master/docs/sources/packages_api/runtime/backendsrvrequest.md) when you use [BackendSrv](https://grafana.com/docs/grafana/latest/packages_api/runtime/backendsrv).
+
+The cancellation logic is:
+- When an ongoing request discovers that an additional request with the same `requestId` has started, then Grafana will cancel the ongoing request.
+- When an ongoing request discovers that the “cancel all requests” `requestId`, then Grafana will cancel the ongoing request.
+
+#### In `Grafana 7.2`
+With Grafana 7.2 we introduced an additional way of canceling requests using [RxJs](https://github.com/ReactiveX/rxjs). To support the new cancellation functionality, the data source needs to use the new `fetch` function in [BackendSrv](https://grafana.com/docs/grafana/latest/packages_api/runtime/backendsrv).
+
+Migrating the core data sources to the new `fetch` function [is an ongoing process that you can read about in this issue.](https://github.com/grafana/grafana/issues/27222)
+
+
+## Request `queue`
+
+### The challenge
+Depending on how web browsers implement the `http1.1` protocol, the browser limits the number of concurrent requests to x number of parallel requests. 
```

Comment:
```suggestion
Depending on the web browser implement the protocol for HTTP 1.1, it may limit the number of concurrent requests. 
```

I think "to x number of parallel requests" is redundant here, but feel free to leave it if you think it's not.